search/constraint.go

```diff
@@ -33,9 +33,9 @@ type Constraint interface {
 	fmt.Stringer
 
 	// filter returns a set of non-overlapping increasing row indexes that may satisfy the constraint.
-	filter(ctx context.Context, rg parquet.RowGroup, primary bool, rr []RowRange) ([]RowRange, error)
+	filter(ctx context.Context, rgIdx int, primary bool, rr []RowRange) ([]RowRange, error)
```

Comment:
I would like to keep interfaces the same... To request the caller to provide the RG instead of assuming the input is `ParquetShard` and call `s.LabelsFile()` explicitly. 

This makes the Constraint code reusable by external projects. The source parquet files don't need to be the same as the parquet shard format (labels and chunk files).

Reply:
What if i keep receiving the `rgIdx int` but the init still be "storage.ParquetFile" ?

Reply:
PTAL?

Reply:
looks good to me. As long as we don't have hard dependency on ParquetShard we should be good.